x/concentrated-liquidity/swaps.go

```diff
@@ -142,6 +142,13 @@ func (k Keeper) SwapExactAmountIn(
 		return osmomath.Int{}, err
 	}
 
+	// Trigger before hook for SwapExactAmountIn prior to mutating state.
+	// If no contract is set, this will be a no-op.
```

Comment:
Seems like the hooks (the contracts) can cause errors as well, can we also add a test case where the contract errors and correctly test if the error in hooks works as expected?

Reply:
Hmm yeah this is a good point. Writing a new CW test contract will require a relatively large iteration on this PR and will also require changes to downstream PRs, so I created an issue here: #6873

I've added the issue to v21 scoping issue to ensure it gets prioritized